### ISSUE TITLE
Code Analysis

### ISSUE THREAD

**Issue by sils1297** [liberapay/liberapay.com#312]
Hey,

we're coala and are using your service. Would you be interested in getting static code analysis for your whole codebase? We can help you improving your code style in one go (coala can do some fixes automatically) and/or improving code style and quality incrementally (GitMate is a code review bot we're building that will only show new issues).

**Comment by Changaco** [comment]
PRs to improve Bug Prevention are very welcome. :-) We currently only have pyflakes partially covering our codebase, it would be great if we could also analyze simplates (#122), and our JavaScript code.

**Comment by sils1297** [comment]
CC @sims1253 @SanketDG

**Comment by Changaco** [comment]
A useful analysis of JavaScript would be undefined variable detection (#100). We'd need to analyze the entire JS code at once for that, not each file individually.

**Comment by sils1297** [comment]
Yeah I looked into that but our routines are too stupid for your setup without anything custom. For that we'd need to compile it together, then do the analysis I guess?

**Comment by Changaco** [comment]
We have a python function that compiles our JS and CSS, we could call it before coala, and point the analyzer to `www/assets/*.js`.

**Comment by sils1297** [comment]
Cool, what command is it? Is there a make target for that?

**Comment by Changaco** [comment]
We don't have a command to compile the assets, and the python code automatically cleans them up on exit so we'll need to tweak that.

**Comment by sils1297** [comment]
Ok, which function is it and which function cleans it up? We can do a make target for coala that compiles it and then runs the analysis.

**Comment by Changaco** [comment]
It's all in `liberapay/utils/http_caching.py`. The `compile_assets` function automatically removes files that didn't exist before it created them. I think we should stop doing that and add a `atexit.register(lambda: clean_assets(www_root))` line back to `liberapay/testing/__init__.py` where it used to be (eef2932332f0ae5cd08815c6b19ad3872c762804).
